pref/qho.jl

```diff
@@ -0,0 +1,76 @@
+using QuBase
+using QuDynamics
+using BenchmarkLite
+
+type SampleProc{Alg} <: Proc end
+
+type Qexpmv end
```

Comment:
@acroy  sorry but I could not understand the construct ... In the example [here](https://github.com/lindahua/BenchmarkLite.jl/blob/master/examples/mathfuns_benchmark.ipynb) the Op is passed as an  parameter, so I thought of doing it this way ... 

Also I tried something on these lines 
```julia
type QuExpmV end

function operator_set(p::SampleProc, cfg::HarmonicOscillator)
    a = lowerop(cfg.N)
    hamiltonian = a'*a + cfg.lambda*(a + a')
    init_state = complex(statevec(1,FiniteBasis(cfg.N)))
    return full(hamiltonian), init_state, a+a'
end


function Base.run{Alg}(p::SampleProc{Alg}, cfg::HarmonicOscillator, s::(QuBase.QuArray, QuBase.QuArray, QuBase.QuArray))
    qprop = QuPropagator(s[1], s[2], cfg.tlist, Alg())
    for (t, psi) in qprop
        psi'*s[3]*psi
    end
end

procs = Proc[ SampleProc{QuExpmV}()]
```

and then running with the `cfgs` I end with an error which says there is no method matching the construct QuPropagator 

```julia
Benchmarking quexpmv ...
ERROR: `QuPropagator{QPM<:QuPropagatorMethod,QVM<:Union(AbstractQuArray{B<:AbstractBasis{S<:AbstractStructure},T,2},AbstractQuArray{B<:AbstractBasis{S<:AbstractStructure},T,1}),QE<:QuEquation}` has no method matching QuPropagator{QPM<:QuPropagatorMethod,QVM<:Union(AbstractQuArray{B<:AbstractBasis{S<:AbstractStructure},T,2},AbstractQuArray{B<:AbstractBasis{S<:AbstractStructure},T,1}),QE<:QuEquation}(::QuArray{FiniteBasis{Orthonormal},Float64,2,Array{Float64,2}}, ::QuArray{FiniteBasis{Orthonormal},Complex{Float64},1,Array{Complex{Float64},1}}, ::FloatRange{Float64}, ::QuExpmV)
```